sdk/samples/iot/README.md

```diff
@@ -684,6 +694,19 @@ Start using the Azure Embedded C SDK IoT Clients in your solutions!
 
 This project welcomes contributions and suggestions. Find more contributing details [here](https://github.com/Azure/azure-sdk-for-c/blob/main/CONTRIBUTING.md).
 
+If you wish to contribute with a sample please read the guideline below:
```

Comment:
A few questions / points here:
1 - I think we should encourage people to reach out to us, via a GitHub issue, *before* creating a new sample so that we can at least give some high-level guidance.  I would hate for someone to spend 2 weeks on a sample that doesn't make sense (e.g. we ourselves days away from creating one).
2 - Does this section belong in this part of the document?  Most of the samples that this readme covers are about "How to connect using C2D messages" level functionality e.g., but I think(?) intent of this is around community samples of "Here's how you run our code on platform-abcd".
3 - Related to this belonging here, this page is quite long already and the vast majority of people using this won't want to create a brand new sample but will have to read through this before getting to the brininging up a sample they care about.

Reply:
thanks for the feedback, John.
1 - Great point. I'll add that as the first bullet point. 
2 - I added it here because it also mentions the sample list, but I disliked it too, so I'll just add a link to the guidelines section of the top readme instead
3 - Oh come on! :-) these are just a few extra lines! the bulk of the text was at the bottom of the page along with the contribution part. But, given #2, I'll remove it anyway so we're good.